checker/tests/index/Issue2334.java

```diff
@@ -0,0 +1,23 @@
+// Test case for issue #2334: http://tinyurl.com/cfissue/2334
```

Comment:
The reference to #2334 was rather confusing. This PR seems to solve a different problem: information about immutable types can be maintained across modifications.
How about modifying this test case and using a general `sideeffecting()` method call instead of the `System.out.println` calls and adding a separate test case that uses `System.out.println`, but uses a mutable type instead of the immutable `String`. That second test is what #2334 is actually about.

Reply:
This solves #2334.  There are other ways to do so, but I don't think we can say which way #2334 is actually about.  I expanded the test case in the way you suggested, and we could create a new issue to focus on other improvements.